torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -540,7 +537,12 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       "ProcessGroupNCCL is only supported with GPUs, no GPUs found!");
   blockingWait_ = parseEnvVarFlag(NCCL_BLOCKING_WAIT);
   asyncErrorHandling_ = parseEnvVarFlag(NCCL_ASYNC_ERROR_HANDLING);
-  desyncDebug_ = parseEnvVarFlag(NCCL_DESYNC_DEBUG);
+  // Infer desync debug from whether TORCH_DISTRIBUTED_DEBUG >= INFO
+  desyncDebug_ = dist_debug_level_ >= DistributedDebugLevel::INFO ? true : false;
+  if (!desyncDebug_) {
+    // Provide backward support of NCCL_DESYNC_DEBUG
+    desyncDebug_ = parseEnvVarFlag(NCCL_DESYNC_DEBUG);
+  }
```

Comment:
`desyncDebug_ = dist_debug_level_ >= DistributedDebugLevel::INFO || parseEnvVarFlag(NCCL_DESYNC_DEBUG)` to do it in one go?

Reply:
Yes, will merge